src/phoenix/db/models.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+from sqlalchemy import (
+    JSON,
+    DateTime,
+    ForeignKey,
+    UniqueConstraint,
+)
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    WriteOnlyMapped,
+    mapped_column,
+    relationship,
+)
+
+
+class Base(DeclarativeBase):
+    type_annotation_map = {
+        Dict[str, Any]: JSON,
+        List[Dict[str, Any]]: JSON,
+    }
+
+
+class Project(Base):
+    __tablename__ = "projects"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str]
+    description: Mapped[Optional[str]]
+    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
+    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
+
+    traces: WriteOnlyMapped["Trace"] = relationship(
+        "Trace",
+        back_populates="project",
+    )
+    __table_args__ = (
+        UniqueConstraint(
+            "name",
+            name="project_name_unique",
+            sqlite_on_conflict="IGNORE",
+        ),
+    )
+
+
+class Trace(Base):
+    __tablename__ = "traces"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    project_rowid: Mapped[int] = mapped_column(ForeignKey("projects.id"))
```

Comment:
Any idea if delete cascade can be setup on these?

Reply:
I added (delete-orphan) cascade to the (one-to-many) relationship.